src/js/reducers/adAccounts.js

```diff
@@ -0,0 +1,20 @@
+import { REQUEST_FETCH_ADACCOUNTS, COMPLETE_FETCH_ADACCOUNTS, FETCH_ADACCOUNTS, SELECT_ADACCOUNT } from '../actions';
+
+const initialState = { all: {}, current: null, isFetching: false };
+
+const adAccounts = (state = initialState, action) => {
+  switch (action.type) {
+    case REQUEST_FETCH_ADACCOUNTS:
+      return { ...state, all: {}, current: null, isFetching: true };
+    case COMPLETE_FETCH_ADACCOUNTS:
+      return { ...state, isFetching: false };
+    case FETCH_ADACCOUNTS:
+      return { ...state, all: action.payload.data.reduce((h, x) => Object.assign(h, { [x.id]: x }), {}) };
```

Comment:
I recommend you use instead Object.assign({}, foo, bar ...) as immutable. Array#reduce() returns a created new array. But this code mutate the original array.